pymeasure/instruments/keysight/keysightE3642A.py

```diff
@@ -0,0 +1,67 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2024 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+from pymeasure.instruments import SCPIMixin, Instrument
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+
+class KeysightE3642A(SCPIMixin, Instrument):
```

Comment:
We already have several Keysight power supplies, is your power supply similar to one of the other ones?
If that's the case, we could simplify the codebase by having just one code for several instruments.

Reply:
What would having just one code for several instruments entail?  Would E3642A not be added to PyMeasure at all? 
 Or would it get added and just inherit functionality from a similar power supply?
 
There are 3 other Keysight power supplies: E3631A, E36312A, and N5767A.  The first two are triple output supplies, whereas my E3642A addition is not.  The N5767A has a voltage range from 0-60 V, whereas the E3642A has a range from 0-8 or 0-20V.  My additions also have different commands from the other supplies.

Reply:
The first goal is, to make that instrument work in pymeasure. 

How a similar instrument is implemented depends on the naming scheme and the similarity.
It could be a series name like "345x series", if the code is exactly the same and the names are from the same series (not here, I think).
In another case it is just inheriting.

If only slight differences exist (like different voltage ranges), you can inherit the instrument and adjust the range with dynamic properties, see https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments/properties.html#instruments-with-similar-features . (Note the `dynamic=True` at the property definition) 

Now for your case:
Thanks for checking the differences.
The most important question is, whether the features are similar and the commands are the same.
One option could be to make your instrument a single channel version.

If you have questions, don't hesitate to ask.

Reply:
I had a look into the different drivers.
Only `KeysightN5767A` seems similar, however, it is in a not that good state (truncated validators, naming..).

A question out of curiosity: would that driver work for your device as well (with the `voltage` / `current` commands, the status...)?

Can your device read the current voltage/current (like the N5767A)?

Reply:
I was unable to test whether the N5767A driver works for my device because the N5767A driver wouldn't accept my PrologixAdapter.  Dunder init in N5767A is looking for a `config` method but the adapter has no such method.

`if isinstance(self.adapter, VISAAdapter):
            self.adapter.config(
                is_binary=False,
                datatype='float32',
                converter='f',
                separator=','
            )`

Reply:
After commenting out the `self.adapter.config` line from `__init__`, I found that the N5767A driver does indeed work for my device.

Reply:
Thanks for checking.